napari/_qt/menus/_tests/test_file_menu.py

```diff
@@ -31,46 +33,119 @@ def _(path):
         uri='some-path/some-file.tif',
     )
     tmp_plugin.manifest.contributions.sample_data = [my_sample]
-
-    viewer = make_napari_viewer()
-    sample_action = viewer.window.file_menu.open_sample_menu.actions()[0]
```

Comment:
Note that we no longer create `open_sample_menu` (like we did in the old `napari/_qt/menus/file_menu.py`). We could still create one and build a `QModelMenu` for samples menu but going with YAGNI for now, as we can use `app` in the test to access the sample menu.

Reply:
For future reference, it seems that you cannot (easily) build a `QModelMenu` for submenus because during teardown error results:

```
_____________________________ ERROR at teardown of test_open_with_dialog_choices_persist ______________________________
  
  >   ???
  
  src\psygnal\_signal.py:980: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
  src\psygnal\_weak_callback.py:367: in cb
      ???
  D:\tmp\.tox\py38-windows-pyqt5\lib\site-packages\app_model\backends\qt\_qmenu.py:124: in _on_registry_changed
      self.rebuild()
  D:\tmp\.tox\py38-windows-pyqt5\lib\site-packages\app_model\backends\qt\_qmenu.py:105: in rebuild
      _rebuild(
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
  
  menu = <app_model.backends.qt._qmenu.QModelMenu object at 0x000001FDE18EE1F0>, app = Application('test_app')
  menu_id = <MenuId.FILE_SAMPLES: 'napari/file/samples'>, include_submenus = True, exclude = None
  
      def _rebuild(
          menu: QMenu | QToolBar,
          app: Application,
          menu_id: str,
          include_submenus: bool = True,
          exclude: Optional[Collection[str]] = None,
      ) -> None:
          """Rebuild menu by looking up `menu` in `Application`'s menu_registry."""
  >       menu.clear()
  E       RuntimeError: wrapped C/C++ object of type QModelMenu has been deleted
  
  D:\tmp\.tox\py38-windows-pyqt5\lib\site-packages\app_model\backends\qt\_qmenu.py:315: RuntimeError
  
  The above exception was the direct cause of the following exception:
  
  _npe2pm = <npe2._pytest_plugin.TestPluginManager object at 0x000001FDE26B3160>
  
      @pytest.fixture
      def builtins(_npe2pm: TestPluginManager):
          with _npe2pm.tmp_plugin(package='napari') as plugin:
  >           yield plugin
  
  napari\conftest.py:280: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
  D:\tmp\.tox\py38-windows-pyqt5\lib\site-packages\npe2\_dynamic_plugin.py:132: in __exit__
      self.cleanup()
  D:\tmp\.tox\py38-windows-pyqt5\lib\site-packages\npe2\_dynamic_plugin.py:90: in cleanup
      self.plugin_manager.unregister(self.manifest.name)
  D:\tmp\.tox\py38-windows-pyqt5\lib\site-packages\npe2\_plugin_manager.py:364: in unregister
      self.deactivate(key)
  D:\tmp\.tox\py38-windows-pyqt5\lib\site-packages\npe2\_plugin_manager.py:433: in deactivate
      ctx._dispose()
  D:\tmp\.tox\py38-windows-pyqt5\lib\site-packages\npe2\_plugin_manager.py:713: in _dispose
      self._disposables.pop()()
  D:\tmp\.tox\py38-windows-pyqt5\lib\site-packages\app_model\registries\_menus_reg.py:68: in _dispose
      self.menus_changed.emit(changed_ids)
  src\psygnal\_signal.py:935: in emit
      ???
```

my guess would be that a submenu has already been destroyed along with the parent menu, so at this step we get this error.